node/network/dispute-distribution/src/sender/mod.rs

```diff
@@ -0,0 +1,361 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::collections::hash_map::Entry;
+
+use futures::Future;
+use futures::channel::mpsc;
+use futures::channel::oneshot;
+use futures::future::RemoteHandle;
+
+use polkadot_node_network_protocol::IfDisconnected;
```

Comment:
You know that you can combine imports? :see_no_evil:

Reply:
I know that, rust-analyzer used to know that as well, but seems to have forgotten with some update  (Using add missing import on auto complete here). Either way, I will clean that up when I am done.

Reply:
rust-analyzer has an `ImportGranularity` config setting, which you might want to set to `module`.

Reply:
rust-analyzer has an `Import Granularity` config item. For me, `preserve` works best. This will just follow along with the current structure.